### PR TITLE
Glimmer conversion progress, changed resize addon

### PR DESCRIPTION
**Pull request by napafundi**

### CODE REVIEW

**Comment by billdami** [comment]
@npafundi don’t forget we can remove the ember-decorators package now that we removed the @observes usage 🎉

File: addon/components/table.hbs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        {{~if this.hasHiddenColumns " has-hidden-columns"}}
+        {{~if @isServerRendered " is-server-rendered"}}
+        {{~if this.isColumnsPanned " is-columns-panned"}}"
+    id={{this.elementId}}
+    ...attributes
+>
+    <div
+        {{did-insert this.didInsertTable}}
+        {{did-resize this.debouncedRender debounce=this.resizeDebounce}}
+        {{did-update this.debouncedRender @columns}}
```
Comment (billdami) [inline addon/components/table.hbs:13]: `did-insert` and `did-update` should probably go on the parent container <div>

File: addon/components/table.ts [hunk 2]
```diff
@@ -1,110 +1,243 @@
...
      */
-    @empty('rows') noRows!: boolean;
+    @computed('tableHeight')
+    get height(): SafeString {
+        return htmlSafe(this.tableHeight ? `height: ${this.tableHeight};` : '');
+    }
+
+    get resizeDebounce(): number {
+        return this.args.resizeDebounce ?? 250;
+    }
```
Comment (billdami) [inline addon/components/table.ts:115]: I'd love if we could eventually turn this pattern into a decorator or something, e.g.
```ts
@argDefault('resizeDebounce') resizeDebounce = 250;
```
maybe something to experiment with if you get some down time 😄

File: addon/components/table.ts [hunk 3]
```diff
@@ -197,8 +322,8 @@ class TableComponent extends Component.extend({ ResizeAware }) {
         //fixed columns are disabled if the widest non-fixed column cannot
         //fit in the container at the same time as the fixed column(s)
         if (!isEmpty(this.fixedColumns)) {
-            const sortedColumns = get(this, 'nonFixedColumns').sortBy('staticWidth');
-            const widestColumn = get(sortedColumns, 'lastObject');
+            const sortedColumns = this.nonFixedColumns.sortBy('staticWidth');
+            const widestColumn = sortedColumns.slice(-1)[0];
```
Comment (billdami) [inline addon/components/table.ts:326]: `const widestColumn = sortedColumns[sortedColumns.length - 1];` is probably a little better here

File: addon/components/table.ts [hunk 4]
```diff
@@ -218,8 +343,8 @@ class TableComponent extends Component.extend({ ResizeAware }) {
      */
     @computed('hasHiddenColumns', 'columns.lastObject.{id,valuePath}', 'visibleColumns.lastObject.{id,valuePath}')
     get canPanRight(): boolean {
-        const lastColId = this.getColumnId(get(get(this, 'columns'), 'lastObject'));
-        const visibleColId = this.getColumnId(get(get(this, 'visibleColumns'), 'lastObject'));
+        const lastColId = this.getColumnId(this.args.columns.slice(-1)[0]);
+        const visibleColId = this.getColumnId(this.visibleColumns.slice(-1)[0]);
```
Comment (billdami) [inline addon/components/table.ts:347]: same for these, use length - 1 to get the last element

File: addon/components/table.ts [hunk 5]
```diff
@@ -383,39 +473,29 @@ class TableComponent extends Component.extend({ ResizeAware }) {
      * @memberof TableComponent
      */
     updateColumnVisibility() {
-        const columns = this.columns || A();
-        const visibleColumns: NativeArray<TableColumn> = A();
-        const containerWidth = this.getElementWidth(this.element);
+        const columns = this.args.columns || A();
+        const visibleColumns: NativeArray<ColumnValue> = A();
+        const containerWidth = this.getElementWidth(window.document.querySelector(`#${this.elementId}`)!);
```
Comment (billdami) [inline addon/components/table.ts:478]: we should probably store the container element once in the `didInsertTable()` call, and you can use `getElementById` here as it'll be a little faster:
```ts
didInsertTable() {
    this.containerElement = document.getElementById(this.elementId);
    //...
}
```

**Comment by billdami** [comment]
@napafundi first pass complete

**Comment by napafundi** [comment]
@billdami this should be good for a final pass

**Comment by billdami** [comment]
@napafundi i think this looks good to go 👍  you still have it marked as a WIP PR tho :-P 

Once you unmark that I'll merge it in and we can publish a new release. (and remember to stop the ember-release process before the last step to do the manual `npm publish` so the TS types are properly compiled)

**Comment by napafundi** [comment]
@billdami WIP removed 😄

**Comment by billdami** [comment]
@napafundi also, in the publish i would bump the minor version, so => 0.4.0, since its a pretty hefty internal refactor.

**Pull request merged**